cli/specifier_handler.rs

```diff
@@ -204,27 +204,16 @@ pub struct FetchHandler {
 }
 
 impl FetchHandler {
-  pub fn new(flags: &Flags, permissions: &Permissions) -> Result<Self> {
+  pub fn new(global_state: &Arc<GlobalState>) -> Result<Self> {
     let custom_root = env::var("DENO_DIR").map(String::into).ok();
     let deno_dir = DenoDir::new(custom_root)?;
-    let deps_cache_location = deno_dir.root.join("deps");
-    let http_cache = HttpCache::new(&deps_cache_location);
-    let ca_file = flags.ca_file.clone().or_else(|| env::var("DENO_CERT").ok());
-
-    let file_fetcher = SourceFileFetcher::new(
-      http_cache,
-      !flags.reload,
-      flags.cache_blocklist.clone(),
-      flags.no_remote,
-      flags.cached_only,
-      ca_file.as_deref(),
-    )?;
     let disk_cache = deno_dir.gen_cache;
+    let file_fetcher = global_state.file_fetcher.clone();
 
     Ok(FetchHandler {
       disk_cache,
       file_fetcher,
-      permissions: permissions.clone(),
+      permissions: Permissions::allow_all(),
```

Comment:
Careful - this might be permission escape - `permissions` should still be passed down from `GlobalState`

Reply:
I thought I tried that and it broke things, like required you to use `--allow-read` to even read a local module. I will try again though.

Reply:
This constructor is called during `prepare_module_load` - so the permissions passed from `prepare_module_load` will definitely be the proper ones.

Reply:
Ah, I think that was the problem, I was cloning them off of the `global_state`... 🤦